### PR TITLE
Remove prometheus extension - ICP-7535

### PR DESCRIPTION
**Pull request by VLazarevic**
## Description

https://dt-rnd.atlassian.net/browse/ICP-7535

Removes the `.spec.extensions.prometheus` feature completely. It never went GA, so no deprecation path.

What changed:
- Dropped the `Prometheus` field / `PrometheusSpec` type and the `IsPrometheusEnabled` helpers, regenerated the CRDs (the `prometheus` property is gone from the extensions schema).
- The OTel collector StatefulSet is now built only for `telemetryIngest`, removed the prometheus-only envs, volumes, args and the extensions-secret-hash annotation. The `telemetryIngest` side is kept intact.
- Removed the `PrometheusExtensions` token feature and the prometheus check in validation.
- Helm: deleted the `extensions/prometheus` templates + tests and the `rbac.extensions.prometheus` block.
- Tests/E2E: dropped the `WithExtensionsPrometheusEnabledSpec` helper; tests that used prometheus only to enable extensions now use `databases`.

Things worth a closer look in review:
- The `trustedCAs` env/volume that was gated only on prometheus is gone, so the "with trusted CAs" cases in the extensions+telemetry+in-cluster-AG volume test now expect it to be absent.
- Kept shared helpers/consts (`GetTLSSecretName`, `GetTokenSecretName`, `ExtensionsTokensVolumeName`, `AnnotationExtensionsSecretHash`) since the EEC/databases/ActiveGate paths still use them.

## How can this be tested?

It's a removal, so nothing new to enable. To verify:
- `go build ./...`, `go test ./pkg/... ./cmd/...`, `make test/helm/unit`.
- Check that `prometheus` is gone under `spec.extensions` in the generated DynaKube CRD, while the `DTPrometheus` CRD is unchanged.
- A DynaKube with `spec.extensions.prometheus` set: the field is now unknown. Extensions still work via `spec.extensions.databases`, and the OTel collector still starts with `telemetryIngest`.

### CODE REVIEW

File: pkg/api/validation/dynakube/testdata/latest-default.yaml [hunk 1]
```diff
@@ -60,8 +60,7 @@ spec:
   customPullSecret: pull-secret
   dynatraceApiRequestThreshold: 42
   enableIstio: false
-  extensions:
```
Comment (avorima) [inline pkg/api/validation/dynakube/testdata/latest-default.yaml:63]: the extensions should not be included in the YAML here. it makes it look like the feature is enabled

File: pkg/api/validation/dynakube/extensions_test.go
Comment (avorima) [inline pkg/api/validation/dynakube/extensions_test.go]: i only introduced this to reduce boilerplate, but it's actually breaking editor integration. please remove this in favor or just running the test cases directly

File: pkg/controllers/dynakube/otelc/statefulset/container.go [hunk 2]
```diff
@@ -32,8 +27,6 @@ func getContainer(dk *dynakube.DynaKube, replicas int32) corev1.Container {
 	}
 
 	// Only enable the probes when we control the configuration.
```
Comment (avorima) [inline pkg/controllers/dynakube/otelc/statefulset/container.go]: the if condition is obsolete actually. go over all `TelemetryIngest().IsEnabled()` and see if they're still needed

**Review by avorima — COMMENTED** [review]
there is a potential issue with the OTEL image defaulting we do during conversion. if a user is still on v1beta5 we default the image in the ref on conversion, but that info is not cleared when there's no telemetryIngest enabled. The removal should handle that edge case

**Comment by avorima** [comment]
the e2e tests also need to be updated, some of them don't actually deploy the EEC anymore

File: pkg/api/validation/dynakube/extensions_test.go
Comment (VLazarevic) [reply pkg/api/validation/dynakube/extensions_test.go]: [f523a0e](https://github.com/Dynatrace/dynatrace-operator/pull/7300/commits/f523a0eb55d22cf02276faa96d8f1f9bfee5b99a)

File: pkg/controllers/dynakube/otelc/statefulset/container.go (hunk 2 above)
Comment (VLazarevic) [reply pkg/controllers/dynakube/otelc/statefulset/container.go]: [953f6a1](https://github.com/Dynatrace/dynatrace-operator/pull/7300/commits/953f6a141abd6fca85862815630dec7020560961)

File: pkg/api/validation/dynakube/testdata/latest-default.yaml (hunk 1 above)
Comment (VLazarevic) [reply pkg/api/validation/dynakube/testdata/latest-default.yaml:63]: [f523a0e](https://github.com/Dynatrace/dynatrace-operator/pull/7300/commits/f523a0eb55d22cf02276faa96d8f1f9bfee5b99a) & [953f6a1](https://github.com/Dynatrace/dynatrace-operator/pull/7300/commits/953f6a141abd6fca85862815630dec7020560961)

File: pkg/api/validation/dynakube/testdata/v1beta5-default.converted.yaml [hunk 3]
```diff
@@ -59,7 +59,6 @@ spec:
   customPullSecret: pull-secret
   dynatraceApiRequestThreshold: 42
   enableIstio: false
-  extensions: {}
```
Comment (avorima) [inline pkg/api/validation/dynakube/testdata/v1beta5-default.converted.yaml:62]: we should just remove the extensions structs from v1beta5 as well. this roundtrip now (correctly) removes this field from the YAML and the controller for it is also gone

File: test/e2e/features/extensions/upgrade/upgrade.go
Comment (avorima) [inline test/e2e/features/extensions/upgrade/upgrade.go]: this whole test has lost its usefulness. there should be no EEC running at all anymore. remove this file and everything that calls it

File: test/e2e/features/extensions/extensions.go
Comment (avorima) [inline test/e2e/features/extensions/extensions.go:1]: this duplicates the dbexecutor test. remove it

File: test/e2e/features/supportarchive/support_archive.go [hunk 4]
```diff
@@ -116,7 +115,6 @@ func Feature(t *testing.T) features.Feature {
 	// check if components are running
 	builder.Assess("active gate pod is running", k8sstatefulset.IsReady(testDynakube.Name+"-"+agconsts.MultiActiveGateName, testDynakube.Namespace))
 	builder.Assess("extensions execution controller started", k8sstatefulset.IsReady(testDynakube.Extensions().GetExecutionControllerStatefulsetName(), testDynakube.Namespace))
```
Comment (avorima) [inline test/e2e/features/supportarchive/support_archive.go:120]: this test case should fail, because there's no extension enabled. need to enable dbextensions

**Review by avorima — COMMENTED** [review]
make sure to run e2e tests

File: test/e2e/features/supportarchive/support_archive.go (hunk 4 above)
Comment (VLazarevic) [reply test/e2e/features/supportarchive/support_archive.go:120]: [5dc3a27](https://github.com/Dynatrace/dynatrace-operator/pull/7300/commits/5dc3a27df769785efbb7d11811a4a07af0e4c762)

File: test/e2e/features/extensions/extensions.go
Comment (VLazarevic) [reply test/e2e/features/extensions/extensions.go:1]: [5dc3a27](https://github.com/Dynatrace/dynatrace-operator/pull/7300/commits/5dc3a27df769785efbb7d11811a4a07af0e4c762)

File: test/e2e/features/extensions/upgrade/upgrade.go
Comment (VLazarevic) [reply test/e2e/features/extensions/upgrade/upgrade.go]: [5dc3a27](https://github.com/Dynatrace/dynatrace-operator/pull/7300/commits/5dc3a27df769785efbb7d11811a4a07af0e4c762)

File: pkg/api/validation/dynakube/testdata/v1beta5-default.converted.yaml (hunk 3 above)
Comment (VLazarevic) [reply pkg/api/validation/dynakube/testdata/v1beta5-default.converted.yaml:62]: [5dc3a27](https://github.com/Dynatrace/dynatrace-operator/pull/7300/commits/5dc3a27df769785efbb7d11811a4a07af0e4c762)

File: test/e2e/features/supportarchive/support_archive.go (hunk 4 above)
Comment (avorima) [reply test/e2e/features/supportarchive/support_archive.go:120]: that's not what i meant. we do want the EEC running for this test

File: test/e2e/scenarios/release/release_test.go [hunk 5]
```diff
@@ -41,11 +39,6 @@ func TestMain(m *testing.M) {
 	testEnv.Setup(helpers.SetScheme)
 
 	testEnv.BeforeEachTest(func(ctx context.Context, envConfig *envconf.Config, t *testing.T) (context.Context, error) {
-		// TODO Remove this after 1.10 release
-		if tenant.UsePlatformToken() {
```
Comment (avorima) [inline test/e2e/scenarios/release/release_test.go:45]: i think this broke e2e tests

Comment (VLazarevic) [reply test/e2e/scenarios/release/release_test.go:45]: I am going to look into it, I've seen the TODO and trusted it too much lol

Comment (avorima) [reply test/e2e/scenarios/release/release_test.go:45]: the TODO is wrong. remove it and restore this if block

Comment (VLazarevic) [reply test/e2e/scenarios/release/release_test.go:45]: [3b61d86](https://github.com/Dynatrace/dynatrace-operator/pull/7300/commits/3b61d862d9f08c68696e68dccc3dda7a2fb6af89)

**Review by avorima — APPROVED** [review]
